messaging/src/main/java/org/axonframework/eventhandling/EventTrackerStatus.java

```diff
@@ -74,4 +76,21 @@
      * @return the exception that caused processing to fail, or {@code null} when processing normally
      */
     Throwable getError();
+
+    /**
+     * Return the estimated relative position this tracking token represents for the event tracker status whether this Segment is still replaying.
+     * In case no estimation can be given or no replay an {@code OptionalLong.empty()} will be returned.
+     *
+     * @return the estimated relative position this tracking token represents for the event tracker status whether this Segment is still replaying
+     */
+    OptionalLong getReplayPosition();
+
+    /**
+     * Return the relative position at which a reset was triggered whether this Segment is still replaying.
+     * In case of a replay ended or no replay an {@code OptionalLong.empty()} will be returned.
+     *
+     * @return the relative position at which a reset was triggered whether this Segment is still replaying
+     */
+    OptionalLong getResetTokenPosition();
+
```

Comment:
Can be useful to have here a method that returns the estimated number of events remaining to process.

Reply:
Good point.
This will be part of a future issue.

Reply:
I would prefer to expose 2 (or ultimately 3) positions: 
- current position
- reset position (if still replaying)
- (and later) the position of the head of the stream

Components that then monitor the state of the processor can then calculate the estimated time for catching up with the reset position and/or head. From the framework, I prefer to emit the simple facts, and let observing components do the "magic" math ;-)

Reply:
Sweet plan. That'll leave the `getResetPosition` as is, but will replace the `getReplayPosition()` method for a regular `getPosition`.